Framework/Kernel/src/UsageService.cpp

```diff
@@ -106,15 +129,41 @@ void UsageServiceImpl::registerStartup() {
 
 /** registerFeatureUsage
  */
-void UsageServiceImpl::registerFeatureUsage(const std::string &type,
-                                            const std::string &name,
+void UsageServiceImpl::registerFeatureUsage(const FeatureType &type,
+                                            std::vector<std::string> name,
```

Comment:
I'd set it up this way because of the trimming of the feature names. I was deliberately copying so that I could then trim without modifying the supplied parameter (which seemed a bit of an obscure side effect of the register function). I will drop the trimming which will simplify things + make them const & as you suggest